src/config/mod.rs

```diff
@@ -105,17 +114,24 @@ impl Config {
                     theme: custom.theme.unwrap_or(default.theme),
                     remote_styles: custom.remote_styles.unwrap_or(default.remote_styles),
                     additional_css: custom.additional_css.unwrap_or(default.additional_css),
-                    repository: custom.repository,
+                    repository: Some(
+                        custom
+                            .repository
+                            .unwrap_or_else(|| project.repository.unwrap_or_default()),
+                    ),
                     syntax_theme: custom.syntax_theme.unwrap_or(default.syntax_theme),
                     analytics: custom.analytics,
                     additional_pages: custom.additional_pages,
                     social: custom.social,
+                    artifacts: custom.artifacts,
+                    version: Some(project.version.unwrap_or_default()),
```

Comment:
again why not just let this be an Option? Empty strings aren't terribly useful values here.

Reply:
also changed